packages/flutter_tools/lib/src/template.dart

```diff
@@ -120,57 +115,57 @@ class Template {
     /// expansion on the path itself.
     ///
     /// Returns null if the given raw destination path has been filtered.
-    String renderPath(String relativeDestinationPath) {
-      final Match match = _kTemplateLanguageVariant.matchAsPrefix(relativeDestinationPath);
+    String? renderPath(String relativeDestinationPath) {
+      final Match? match = _kTemplateLanguageVariant.matchAsPrefix(relativeDestinationPath);
       if (match != null) {
-        final String platform = match.group(1);
-        final String language = context['${platform}Language'] as String;
+        final String? platform = match.group(1);
+        final String? language = context['${platform}Language'] as String?;
         if (language != match.group(2)) {
           return null;
         }
-        relativeDestinationPath = relativeDestinationPath.replaceAll('$platform-$language.tmpl', platform);
+        relativeDestinationPath = relativeDestinationPath.replaceAll('$platform-$language.tmpl', platform!);
```

Comment:
Force unwrap `platform` will crash if `platform` is null.

Reply:
`match.group(1)` can't be null if `match` isn't null, it means there was at least one match.  And it's 1 indexed.